cpp/include/cudf/column/column.hpp

```diff
@@ -75,6 +76,30 @@ class column {
    */
   column(column&& other) noexcept;
 
+  /**
+   * @brief Construct a new column by taking ownership of the contents of a device_uvector.
+   *
+   * @param other The device_uvector whose contents will be moved into the new column.
+   * @param[in] null_mask Optional, column's null value indicator bitmask. May
+   * be empty if `null_count` is 0 or `UNKNOWN_NULL_COUNT`.
+   * @param null_count Optional, the count of null elements. If unknown, specify
+   * `UNKNOWN_NULL_COUNT` to indicate that the null count should be computed on
+   * the first invocation of `null_count()`.
+   */
+  template <typename T, CUDF_ENABLE_IF(cudf::is_numeric<T>() or cudf::is_chrono<T>())>
+  column(rmm::device_uvector<T>&& other,
+         rmm::device_buffer&& null_mask = {},
+         size_type null_count           = cudf::UNKNOWN_NULL_COUNT)
+    : _type{cudf::type_to_id<T>()},
+      _size{static_cast<size_type>(other.size())},
+      _data{other.release()},
```

Comment:
This is bad. It invokes the same bug that we just talk about: it is unknown if `other.size()` still returns the valid size or `other` has already been emptied out.

Reply:
See https://github.com/rapidsai/cudf/pull/11356#issuecomment-1196152053

Reply:
I believe `other.size()` is guaranteed to be called before `other.release()` since `_size` is declared before `_data` in the class definition. 

> 3) Then, non-static data member are initialized in order of declaration in the class definition.

from the [constructor ](https://en.cppreference.com/w/cpp/language/constructor) page

Reply:
A constructor from `device_uvector<T>&&` solves this through list initialization sequencing:
> 10) In [list-initialization](https://en.cppreference.com/w/cpp/language/list_initialization), every value computation and side effect of a given initializer clause is sequenced before every value computation and side effect associated with any initializer clause that follows it in the brace-enclosed comma-separated list of initializers.

Reply:
Right. But it is still dangerous if the order of `_size` and `_data` is changed in the future.

Reply:
While it is good practice to write resilient and safe code where possible, developers changing code are responsible for the effects of their changes. This is a reasonable and expected behavior for constructors and member ordering.